src/pages/api/data.js

```diff
@@ -1,8 +1,19 @@
 import { fetchAllNodes } from 'api';
 
 export default async function handler(req, res) {
+  const daysAgo = req.query.length;
```

Comment:
You need a default value in case `req.query.length` is undefined.

Reply:
Fixed